java/s2i/assemble

```diff
@@ -93,17 +114,25 @@ function build_maven() {
   mvn ${maven_env_args} ${maven_args} ${MAVEN_ARGS_APPEND}
   check_error "Maven build" $?
 
-  # ======================
-  # After-build validation
-  if [ ! -d "${build_dir}/hawt-app" ] && [ x"${JAVA_MAIN_CLASS}" = x ]; then
-    echo "No JAVA_MAIN_CLASS specified and no hawt-app-maven-plugin configured in the pom.xml"
-    exit 1
-  fi
-
   # ==============
   # Copy artifacts
-  echo "Copying Maven artifacts from ${build_dir}/hawt-app to ${app_dir} ..."
-  copy_dir "${build_dir}/hawt-app" ${app_dir}
+  if [ -d "${build_dir}/hawt-app" ]; then
+    echo "Hawtapp plugin detected. Copying Maven artifacts from ${build_dir}/hawt-app to ${app_dir} ..."
```

Comment:
Yes, I know.. but it forced the execution of the hawtapp maven goal, that is something you don't desire in non-hawtapp application. I considered that most of the quickstarts and samples have the hawtapp plugin configured, so there is no need to call that goal in the image builder.

Also, you can still use hawtapp by providing specific `MAVEN_ARGS` in the template, even if the pluign is not configured in the pom. The new image just checks the output of the maven build.